### ISSUE TITLE
Problem loading zg for 6hrPlevPt

### ISSUE THREAD

**Issue by sloosvel** [ESMValGroup/ESMValCore#333]
**Describe the bug**

When trying to work with the geopotential height for `mip: 6hrPlevPt`, the variable on BADC (and on ESGF) is called `zg`. But the tool gives the following error:

`esmvalcore._recipe_checks.RecipeError: Unable to load CMOR table 'CMIP6' for variable 'zg' with mip '6hrPlevPt'`

Which is right because there is no such variable called `zg` in this table. Is this an issue with the way the files are named? Or is this something the tool should be able to handle? 

I tried to attach an example recipe but apparently attaching yml files is not supported, here is an example dataset:

 - {project: CMIP6, mip: 6hrPlevPt, dataset: CNRM-CM6-1, exp: historical, ensemble: r1i1p1f2, grid: gr, start_year: 1982, end_year: 1982}

**Comment by zklaus** [comment]
Maybe it has something to do with the fact that in this table there are two variables (`zg7h` and `zg27`) both with the same `out_name` of `zg` that offer essentially the same information but on different levels.

Have a look at PCMDI/cmor#409 and https://github.com/PCMDI/cmor/blob/master/LibCV/PrePARE/out_names_tests.json to see how CMOR/PrePARE deals with that.

**Comment by sloosvel** [comment]
Yes I have noticed that in this table there are several cases in which the variable name differs from the `out_name`. I think that the problem comes from assuming that `var_name = out_name` when loading all the cmor information for the variables? In the VariableInfo class the out_name is not something that gets read.

**Comment by thomascrocker** [comment]
Just to add a +1 to this issue from me. One of the diagnostics I have been asked to implement in ESMValTool for the LaunchPAD project https://www.geog.ox.ac.uk/news/2019/0708-launchPAD.html requires geopotential height (zg) data from the zg27 CMIP6 table. It looks like #1099 should address the problem once it is merged.

**Comment by schlunma** [comment]
Moving this to v2.6 to follow the corresponding PR.

**Comment by sloosvel** [comment]
I don't think I will have time to take care of this for 2.6, sorry. I will remove it from the milestone.

**Comment by valeriupredoi** [comment]
sorry, no way this can be sorted out for M2.7 :cry:

**Comment by valeriupredoi** [comment]
hi @sloosvel how realistic would be a solution for this in v2.9 given code freeze is kicking in on Monday? :beer:

**Comment by valeriupredoi** [comment]
fair enough, will bump to 2.10 then, cheers :beer:
